python/ray/tune/tests/test_tune_restore.py

```diff
@@ -83,6 +85,73 @@ def testPostRestoreCheckpointExistence(self):
         self.assertTrue(os.path.isfile(self.checkpoint_path))
 
 
+class TuneFailResumeTest(unittest.TestCase):
+    class FailureInjectorCallback(Callback):
+        """Adds random failure injection to the TrialExecutor."""
+
+        def __init__(self, steps=20):
+            self._step = 0
+            self.steps = steps
+
+        def on_trial_start(self, trials, **info):
+            self._step += 1
+            if self._step >= self.steps:
+                raise RuntimeError
+
+    class CheckStateCallback(Callback):
+        """Checks state for the experiment initialization."""
+
+        def __init__(self, expected=20):
+            self.expected = expected
+            self._checked = False
+
+        def on_step_begin(self, iteration, trials, **kwargs):
+            if not self._checked:
+                assert len(trials) == self.expected
+                self._checked = True
+
+    def setUp(self):
+        ray.init(num_cpus=2)
+        self.logdir = tempfile.mkdtemp()
+        os.environ["TUNE_GLOBAL_CHECKPOINT_S"] = "0"
+        from ray.tune import register_trainable
+        register_trainable("trainable", MyTrainableClass)
+
+    def tearDown(self):
+        ray.shutdown()
+        os.environ.pop("TUNE_GLOBAL_CHECKPOINT_S")
+        shutil.rmtree(self.logdir)
+        _register_all()
+
+    def testFailResumeGridSearch(self):
+        config = dict(
+            num_samples=5,
+            fail_fast=True,
+            config={
+                "test": tune.grid_search([1, 2, 3]),
+                "test2": tune.grid_search([1, 2, 3]),
+            },
+            stop={"training_iteration": 2},
+            local_dir=self.logdir,
+            verbose=1)
+
+        try:
+            tune.run(
+                "trainable",
+                callbacks=[self.FailureInjectorCallback()],
+                **config)
+            raise Exception("Expected value error!")
```

Comment:
`with self.assertRaises(RuntimeError)`?

Reply:
hm technically not something we test, so maybe semantically not necessary

Reply:
addressed!